content/guides/end-to-end-testing/amazon-cognito-authentication.md

```diff
@@ -157,6 +158,169 @@ const awsConfig = require('./aws-exports-es5.js')
 
 ## Custom Command for Amazon Cognito Authentication
 
+There are two ways you can authenticate to AWS Cognito
+
+- [Login with cy.origin()](/guides/end-to-end-testing/amazon-cognito-authentication#Login-with-cy-origin)
+- [Legacy Programmatic Access](/guides/end-to-end-testing/amazon-cognito-authentication#Programmatic-Login-Legacy)
+
+### Login with [`cy.origin()`](/api/commands/origin)
+
+Next, we'll write a command to perform a login to
+[Amazon Cognito](https://aws.amazon.com/cognito) using
+[`cy.origin()`](/api/commands/origin) to navigate to the cognito origin,
+inputting a user credentials, and signing in via OAuth to redirect back to our
+application, and caching the results with
+[`cy.session()`](/api/commands/session).
+
+In this `loginByCognito` command, we redirect to the cognito login screen by
+authenticating with `Sign in with AWS`. We then enter our user credentials and
+sign in, which redirects us back to the realworld app.
+
+```jsx
+// cypress/support/auth-provider-commands/cognito.ts
+
+/// Amazon Cognito
+Cypress.Commands.add('loginByCognito', (username, password) => {
+  Cypress.log({
+    displayName: 'COGNITO LOGIN',
+    message: [`🔐 Authenticating | ${username}`],
+    // @ts-ignore
+    autoEnd: false,
+  })
+
+  cy.visit('/')
+  cy.contains('Sign in with AWS', {
+    includeShadowDom: true,
+  }).click()
+
+  cy.origin(
+    Cypress.env('cognito_domain'),
+    {
+      args: {
+        username,
+        password,
+      },
+    },
+    ({ username, password }) => {
+      // cognito log in page has some elements of the same id but are off screen. we only want the visible elements to log in
+      cy.get('input[name="username"]:visible').type(username)
+      cy.get('input[name="password"]:visible').type(password, {
+        log: false,
+      })
+      cy.get('input[name="signInSubmitButton"]:visible').click()
+    }
+  )
+
+  // give a few seconds for redirect to settle
+  // eslint-disable-next-line cypress/no-unnecessary-waiting
+  cy.wait(2000)
+
+  // verify we have made it passed the login screen
+  cy.contains('Get Started').should('be.visible')
+})
+```
+
+Secondly, we can use our `loginByCognito` command in the test. Below is our test
+to login as a user via [Amazon Cognito](https://aws.amazon.com/cognito),
+complete the onboarding process and logout.
+
+<Alert type="success">
+
+The
+[runnable version of this test](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui-auth-providers/cognito.spec.ts)
+is in the
+[Cypress Real World App](https://github.com/cypress-io/cypress-realworld-app).
+
+</Alert>
+
+```jsx
+describe('Cognito', function () {
+  beforeEach(function () {
+    // Seed database with test data
+    cy.task('db:seed')
+
+    // login via Amazon Cognito via cy.origin()
+    cy.loginByCognito(
+      Cypress.env('cognito_username'),
+      Cypress.env('cognito_password')
+    )
```

Comment:
ever use the diff markdown language? not sure what we typically do, but could highlight like:
```diff
  beforeEach(function () {
    // Seed database with test data
    cy.task('db:seed')
    // login via Amazon Cognito via cy.origin()
+    cy.loginByCognito(
+      Cypress.env('cognito_username'),
+      Cypress.env('cognito_password')
+    )
```

Reply:
I haven't. That's neat!

Reply:
really? its my fav 😄

Reply:
It does a look a bit weird with the docs markdown renderer. Doesn't look like we get the cool green highlights:
<img width="849" alt="Screen Shot 2022-12-01 at 11 13 22 AM" src="https://user-images.githubusercontent.com/3980464/205103544-049bf1b2-b375-4e39-a165-658e588c4cb6.png">

Reply:
@AtofStryker did you use ` ```js ` or ` ```diff`? looks like it's using JS/jsx

Reply:
it's either JS or diff, not both :/ so JSX might be clearer.

Reply:
might just stick with the jsx on this one. the contract just looks off 
<img width="690" alt="Screen Shot 2022-12-01 at 12 50 19 PM" src="https://user-images.githubusercontent.com/3980464/205124717-deb9d860-5334-49b6-a630-b25b9ddb6f44.png">